src/course-unit/data/slice.js

```diff
@@ -103,6 +103,12 @@ const slice = createSlice({
     fetchStaticFileNoticesSuccess: (state, { payload }) => {
       state.staticFileNotices = payload;
     },
+    reorderXBlockList: (state, { payload }) => {
+      const xBlockList = [...state.courseVerticalChildren.children];
+      xBlockList.sort((a, b) => payload.indexOf(a.id) - payload.indexOf(b.id));
+
+      state.courseVerticalChildren.children = [...xBlockList];
+    },
```

Comment:
```suggestion
reorderXBlockList: (state, { payload }) => {
  // Create a map for payload IDs to their index for O(1) lookups
  const indexMap = new Map(payload.map((id, index) => [id, index]));

  // Directly sort the children based on the order defined in payload
  // This avoids the need to copy the array beforehand
  state.courseVerticalChildren.children.sort((a, b) => {
    return (indexMap.get(a.id) || 0) - (indexMap.get(b.id) || 0);
  });
},
```

`indexOf` inside `sort` function will be ineffective.

Reply:
I moved this code from another page, but your suggestion looks great!